hydra_base/lib/attributes.py

```diff
@@ -449,8 +451,13 @@ def add_attribute(attr, check_existing=True, **kwargs):
 
         if attr.network_id is not None:
             attr_qry = attr_qry.filter(Attr.network_id == attr.network_id)
+        else:
+            attr_qry = attr_qry.filter(Attr.network_id == None)
+
         if attr.project_id is not None:
             attr_qry = attr_qry.filter(Attr.project_id == attr.project_id)
+        else:
+            attr_qry = attr_qry.filter(Attr.project_id==None)
```

Comment:
If `attr.network_id` is `None` here, then the query on 453 is equivalent to that on 455 so the test on 452 can be omitted.

Same for the subsequent `attr.project_id` case.